overlord/devicestate/firstboot_test.go

```diff
@@ -169,18 +181,91 @@ func (s *FirstBootTestSuite) TestPopulateFromSeedOnClassicNoSeedYaml(c *C) {
 
 	tsAll, err := devicestate.PopulateStateFromSeedImpl(st)
 	c.Assert(err, IsNil)
-	// only mark seeded
-	c.Check(tsAll, HasLen, 1)
-	tasks := tsAll[0].Tasks()
-	c.Check(tasks, HasLen, 1)
-	c.Check(tasks[0].Kind(), Equals, "mark-seeded")
+	checkTrivialSeeding(c, tsAll)
 
 	ds, err := auth.Device(st)
 	c.Assert(err, IsNil)
 	c.Check(ds.Brand, Equals, "my-brand")
 	c.Check(ds.Model, Equals, "my-model-classic")
 }
 
+func (s *FirstBootTestSuite) TestPopulateFromSeedOnClassicNoSeedYamlWithCloudInstanceData(c *C) {
+	release.OnClassic = true
```

Comment:
Should we use `restore := release.MockOnClassic(true); defer restore()` here?

Reply:
the test suite is already doing the restore, although a comment would be nice

Reply:
tbh, I just followed the style of some tests already present here